scripts/template_matcher.py

```diff
@@ -7,21 +9,27 @@
 
 class TemplateMatcher(object):
 
-    def __init__ (self, images, min_match_count=10, good_thresh=0.7):
+    def __init__ (self, images, min_match_count=5, good_thresh=0.7):
+        cv2.namedWindow('template_img')
+        cv2.namedWindow('result_img')
         self.signs = {} #maps keys to the template images
         self.kps = {} #maps keys to the keypoints of the template images
         self.descs = {} #maps keys to the descriptors of the template images
-        if cv2.__version__=='3.1.0-dev':
+        if cv2.__version__=='3.1.0-dev' or cv2.__version__=='3.2.0': #this is gonna break in future versions so bad
             self.sift = cv2.xfeatures2d.SIFT_create()
         else:
             self.sift = cv2.SIFT() #initialize SIFT to be used for image matching
 
         # for potential tweaking
         self.min_match_count = min_match_count
         self.good_thresh = good_thresh #use for keypoint threshold
+        self.ransac_thresh = 5.0
 
-        #TODO: precompute keypoints for template images
-
+        for k, filename in images.iteritems():
```

Comment:
I saw another person put the precomputing of keypoints and descriptors in a separate function, i.e.  `precomputeTemplateKpsAndDescs` and then calling this function in the __init__ function.  I thought it was a really good way to keep the level of abstraction in the __init__ the same, while also modularizing different parts of the data pipeline.  Curious if you think that's a good idea!

Credits to Sean:
https://github.com/SeanCCarter/sign_follower/blob/master/scripts/template_matcher.py#L36